### DISCUSSION TITLE
Spaced-repetition

### DISCUSSION THREAD

**Discussion by pnodet** [chesspecker/chesspecker#29] [General]
> Spaced repetition is an evidence-based learning technique that is usually performed with flashcards. Newly introduced and more difficult flashcards are shown more frequently, while older and less difficult flashcards are shown less frequently in order to exploit the psychological spacing effect. The use of spaced repetition has been proven to increase rate of learning.
> _[Wikipedia Spaced repetition](https://en.wikipedia.org/wiki/Spaced_repetition)_

Implementing space repetition would be a big plus.

**Comment by pnodet** [comment]
**Ressources:**

Basic SM2 Algorithm, easy to implement.
[Maxvien/supermemo](https://github.com/Maxvien/supermemo)

SM2+, small improvements, easy to implement.
[A Better Spaced Repetition Learning Algorithm: SM2+](http://www.blueraja.com/blog/477/a-better-spaced-repetition-learning-algorithm-sm2)

Memory Scheduler Algorithm, improved algorithm, easy to implement.
[A Simple but Effective Learning Algorithm: MS Syntax](https://blog.lotp.xyz/2018/08/12/A-Simple-But-Effective-Spaced-Repitition-Algorithm-MS/)

JS implementation of SM-15, improved algorithm, hard to implement.
[slaypni/SM-15](https://github.com/slaypni/SM-15)

**Reply by cshancock** [reply]
I take this to mean that as of right now, there is no spaced repetition algorithm in use? 

I am glad to see you pursue this project. The website is fast and crisp. I have been using the Lichess puzzles database in LucasChess R, but wasn't sure how the spaced repetition functioned in it. When looking for alternatives I found your website.

**Reply by pnodet** [reply]
Spaced implementation will mean a lot of changes. Indeed as of now it is not implemented.
I would like it to be though and I work on it on my free time but can't promise anything!
Thanks for the support!

**Comment by zeezop** [comment]
@pnxdxt: Have you looked at: https://www.chessflashcards.com/create/tactics/ ?

**Reply by pnodet** [reply]
Thanks!

**Comment by b-kaiser** [comment]
May I ask to what extent spaced repetition is currently implemented? I started using it and it seems working, even though the statistics then confuse me and might be broken. Unfortunately, there is very little information about how the spaced repetition works. Why is it always sets of 20?

**Reply by pnodet** [reply]
Hi! Currently there is a first implementation that splits the set in small groups of 20. Roughly, when you complete a group, the puzzles you failed are move down and the puzzle you completed with success move up a group.

**Reply by b-kaiser** [reply]
Hi! Thanks for the quick reply. I hope I don't bother, and by the way: This is an amazing project! To be honest, I do not quite understand. But basically however it works, so far I feel comfortable with the algorithm. However, someday you might want to consider writing a sentence or two about that in some info-section.

**Reply by pnodet** [reply]
Great idea, will do!
